In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
import math
import statistics as stats
import numpy as np
import scipy.stats
import pandas as pd
from statsmodels.stats import weightstats as stests
from scipy.stats import norm
import matplotlib.pyplot as plt
import statsmodels.api as sm
import statsmodels.graphics.gofplots as sm
from statsmodels.stats.weightstats import ztest as ztest
from statsmodels.formula.api import ols
from scipy.stats import f_oneway
import statsmodels.api as sm

In [3]:
data = pd.read_csv('drive/My Drive/Colab/13e2.csv', sep=',', na_values=".")
data

a  b  c  Rep
0  -1 -1 -1  221
1   1 -1 -1  325
2  -1  1 -1  354
3   1  1 -1  552
4  -1 -1  1  440
5   1 -1  1  406
6  -1  1  1  605
7   1  1  1  392
8  -1 -1 -1  311
9   1 -1 -1  435
10 -1  1 -1  348
11  1  1 -1  472
12 -1 -1  1  453
13  1 -1  1  377
14 -1  1  1  500
15  1  1  1  419

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16 entries, 0 to 15
Data columns (total 4 columns):
 #   Column  Non-Null Count  Dtype
---  ------  --------------  -----
 0   a       16 non-null     int64
 1   b       16 non-null     int64
 2   c       16 non-null     int64
 3   Rep     16 non-null     int64
dtypes: int64(4)
memory usage: 640.0 bytes


In [26]:
model = ols('Rep ~ C(a) + C(b) + C(c) + C(a):C(b) + C(b):C(c) + C(a):C(c) + C(a):C(b):C(c)', data).fit()
aov_table = sm.stats.anova_lm(model, typ=3)
aov_table

sum_sq   df          F    PR(>F)
Intercept       141512.00  1.0  57.466802  0.000064
C(a)             12996.00  1.0   5.277563  0.050684
C(b)              7225.00  1.0   2.934010  0.125086
C(c)             32580.25  1.0  13.230558  0.006614
C(a):C(b)         1104.50  1.0   0.448528  0.521892
C(b):C(c)          220.50  1.0   0.089543  0.772385
C(a):C(c)        14280.50  1.0   5.799188  0.042631
C(a):C(b):C(c)    4830.25  1.0   1.961523  0.198923
Residual         19700.00  8.0        NaN       NaN

13.4)

a) Do any of the three factors affect tool life?

Yes if we look at the PR(>F) for letters a and c they are very low numbers compared to the rest. This tells us that a and c which are speed and cutting angle are statistically significant.

b) What combination of factor levels produces the longest tool life?

Th best combinations of factors that produces the longest tool life is a (speed) and c (cutting angle) has their is signifance in the data.

c) Is there a combination of cutting speed and cutting angle that always gives good results regardless of metal hardness?

Yes there is a good combination that gives us a PR(>F) value of 0.042631

In [10]:
data2 = pd.read_csv('drive/My Drive/Colab/13e3.csv', sep=',', na_values=".")
data2

a  b  c  d  Replicate
0  -1 -1 -1 -1        188
1   1 -1 -1 -1        172
2  -1  1 -1 -1        179
3   1  1 -1 -1        185
4  -1 -1  1 -1        175
5   1 -1  1 -1        183
6  -1  1  1 -1        190
7   1  1  1 -1        175
8  -1 -1 -1  1        200
9   1 -1 -1  1        170
10 -1  1 -1  1        189
11  1  1 -1  1        183
12 -1 -1  1  1        201
13  1 -1  1  1        181
14 -1  1  1  1        189
15  1  1  1  1        178
16 -1 -1 -1 -1        195
17  1 -1 -1 -1        180
18 -1  1 -1 -1        187
19  1  1 -1 -1        178
20 -1 -1  1 -1        180
21  1 -1  1 -1        178
22 -1  1  1 -1        180
23  1  1  1 -1        168
24 -1 -1 -1  1        193
25  1 -1 -1  1        178
26 -1  1 -1  1        181
27  1  1 -1  1        188
28 -1 -1  1  1        188
29  1 -1  1  1        173
30 -1  1  1  1        182
31  1  1  1  1        182

In [11]:
data2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32 entries, 0 to 31
Data columns (total 5 columns):
 #   Column     Non-Null Count  Dtype
---  ------     --------------  -----
 0   a          32 non-null     int64
 1   b          32 non-null     int64
 2   c          32 non-null     int64
 3   d          32 non-null     int64
 4   Replicate  32 non-null     int64
dtypes: int64(5)
memory usage: 1.4 KB


In [33]:
model2 = ols('Replicate ~ C(a) + C(b) + C(c) + C(d) + C(a):C(b) + C(b):C(c) + C(c):C(d) + C(a):C(c) + C(a):C(d) + C(b):C(d) + C(a):C(b):C(c) + C(a):C(b):C(d) + C(a):C(c):C(d) + C(b):C(c):C(d) + C(a):C(b):C(c):C(d)', data2).fit()
aov_table2 = sm.stats.anova_lm(model2, typ=3)
aov_table2

sum_sq    df            F        PR(>F)
Intercept            73344.50000   1.0  2537.323243  4.681932e-19
C(a)                   240.25000   1.0     8.311351  1.081710e-02
C(b)                    72.25000   1.0     2.499459  1.334494e-01
C(c)                   196.00000   1.0     6.780541  1.918634e-02
C(d)                    25.00000   1.0     0.864865  3.662006e-01
C(a):C(b)               98.00000   1.0     3.390270  8.419973e-02
C(b):C(c)              128.00000   1.0     4.428108  5.151492e-02
C(c):C(d)               72.00000   1.0     2.490811  1.340769e-01
C(a):C(c)              171.12500   1.0     5.920000  2.707200e-02
C(a):C(d)               24.50000   1.0     0.847568  3.709145e-01
C(b):C(d)                4.50000   1.0     0.155676  6.983790e-01
C(a):C(b):C(c)         232.56250   1.0     8.045405  1.191036e-02
C(a):C(b):C(d)          20.25000   1.0     0.700541  4.149291e-01
C(a):C(c):C(d)          45.56250   1.0     1.576216  2.273330e-01
C(b):C(c):C(d)          45.56250   1.0     1.576216  2.273330e-01
C(a):C(b):C(c):C(d)     47.53125   1.0     1.644324  2.180014e-01
Residual               462.50000  16.0          NaN           NaN

13.6) Analyze the data and draw conclusions

Analyzing the data using two way factorial anova, assuming we are using a 95% confidence level. There is statistical significance in our data set. Looking at the first 4 factors, the following make an impact on the taste of a soft-drink beverage; type of sweetener and carbonation level, each as a PR(>F) value of .0108 and .0192 respectevely. Now if we look at the different combinations of these factors we statistical signifance with the following; a (type of sweetener) and c (carbonation level), a and b (ratio of syrup to water) and c. The rest of the factors and combination are not significant and should not be considered in drawing our conclusions for this data set.